### PR TITLE
Fix Dry::Types deprecations

### PR DESCRIPTION
**Pull request by lostie**
Addresses the following deprecation warnings:

```ruby
[dry-types] Dry::Types.module is deprecated and will be removed in the next major version
Use Dry.Types() instead. Beware, it exports strict types by default, for old behavior use Dry.Types(default: :nominal). See more options in the changelog
```

```
[dry-struct] Dry::Struct::Value is deprecated and will be removed in the next major version
```

**Also:**

- Relaxes the `bundler` version dependency for development to allow for more recent versions to be used

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@
 require 'pry'
 
 # Requires supporting ruby files with custom matchers and macros, etc, in spec/support/ and its subdirectories.
-Dir[File.dirname(__FILE__) + '/support/**/*.rb'].each { |file| require file }
+Dir[File.dirname(__FILE__) + '/support/**/*.rb'].sort.each { |file| require file }
```
Comment (wilsonsilva) [inline spec/spec_helper.rb]: Why is this necessary?

**Comment by wilsonsilva** [comment]
Thanks!

File: spec/spec_helper.rb (hunk 1 above)
Comment (lostie) [reply spec/spec_helper.rb]: If you refer to the commit log, this was changed/fixed as Rubocop flagged it during the TravisCI build (which weirdly should already be failing in master, as my changes didn't affect any of the Rubocop flagged offenses 🤷‍♂ ). 

```ruby
spec/spec_helper.rb:20:1: W: Lint/NonDeterministicRequireOrder: Sort files before requiring them.
Dir[File.dirname(__FILE__) + '/support/**/*.rb'].each { |file| require file }
```
More info on this [here](https://rubocop.readthedocs.io/en/latest/cops_lint/#lintnondeterministicrequireorder).

**Comment by lostie** [comment]
Thank you! :D
